tests/codegen/transmute-optimized.rs

```diff
@@ -92,14 +92,14 @@ pub enum OneTwoThree {
 // CHECK-LABEL: i8 @ordering_transmute_onetwothree(i8
 #[no_mangle]
 pub unsafe fn ordering_transmute_onetwothree(x: std::cmp::Ordering) -> OneTwoThree {
-    // CHECK: ret i8 1
+    // CHECK: ret i8 %x
     std::mem::transmute(x)
 }
```

Comment:
Previously, LLVM would intersect the validity ranges of `Ordering` and `OneTwoThree`, and deduce that the only valid value is `1`. Without the assumes, this no longer happens (although the information is not lost--the argument and return value get `range` attributes, so this could in theory be recovered via a new LLVM opt).

This was added in https://github.com/rust-lang/rust/pull/109993/files#diff-658bfaf1eeca92ac9090a307cc820f2a7afa3c29c047d707a34da3531133f63aR89-R101. This specific example feels pretty obscure, so I don't think this change is significant, unless there's a real-world usecase this is derived from (@scottmcm?)